TR123/Autosplitter.cs

```diff
@@ -0,0 +1,245 @@
+        return passportWasChosen;
+    }
+
+    /// <summary>Determines if the timer should start.</summary>
+    /// <param name="state"><see cref="LiveSplitState" /> passed by LiveSplit</param>
+    /// <returns><see langword="true" /> if the timer should start, <see langword="false" /> otherwise</returns>
+    public bool ShouldStart(LiveSplitState state)
+    {
+        uint oldLevelTime = GameData.LevelIgt.Old;
+        uint currentLevelTime = GameData.LevelIgt.Current;
+
+        // Perform new game logic first, since it is the only place where FG should start.
+        uint currentLevel = GameData.CurrentLevel();
+        bool levelTimeJustStarted = oldLevelTime > 0 && currentLevelTime == 0;
+        bool newGameStarted = levelTimeJustStarted && IsFirstLevel(currentLevel);
+        if (newGameStarted)
+            return true;
+
+        return !Settings.FullGame && levelTimeJustStarted;
+    }
+
+    /// <summary>Determines if <paramref name="level" /> is the first of the game or expansion.</summary>
+    /// <param name="level">Level to check</param>
+    /// <returns><see langword="true" /> if <paramref name="level" /> is the first; <see langword="false" /> otherwise.</returns>
+    public bool IsFirstLevel(uint level) =>
+        CurrentActiveGame switch
+        {
+            Game.Tr1 or Game.Tr1NgPlus => (Tr1Level)level is Tr1Level.Caves,
+            Game.Tr1UnfinishedBusiness => (Tr1Level)level is Tr1Level.AtlanteanStronghold,
+            Game.Tr2 or Game.Tr2NgPlus => (Tr2Level)level is Tr2Level.GreatWall,
+            Game.Tr2GoldenMask         => (Tr2Level)level is Tr2Level.TheColdWar,
+            Game.Tr3 or Game.Tr3NgPlus => (Tr3Level)level is Tr3Level.Jungle,
+            Game.Tr3TheLostArtifact    => (Tr3Level)level is Tr3Level.HighlandFling,
+            _ => throw new ArgumentOutOfRangeException(nameof(CurrentActiveGame), CurrentActiveGame, "Unknown Game"),
+        };
+
+    /// <summary>Determines if <paramref name="level" /> is the last of the game or expansion.</summary>
+    /// <param name="level">Level to check</param>
+    /// <returns><see langword="true" /> if <paramref name="level" /> is the last; <see langword="false" /> otherwise.</returns>
+    public bool IsLastLevel(uint level) =>
+        CurrentActiveGame switch
+        {
+            Game.Tr1 or Game.Tr1NgPlus => (Tr1Level)level is Tr1Level.TheGreatPyramid,
+            Game.Tr1UnfinishedBusiness => (Tr1Level)level is Tr1Level.TempleOfTheCat,
+            Game.Tr2 or Game.Tr2NgPlus => (Tr2Level)level is Tr2Level.HomeSweetHome,
+            Game.Tr2GoldenMask => (Tr2Level)level is Tr2Level.Kingdom or Tr2Level.NightmareInVegas,
+            Game.Tr3 or Game.Tr3NgPlus => (Tr3Level)level is Tr3Level.MeteoriteCavern or Tr3Level.AllHallows,
+            Game.Tr3TheLostArtifact => (Tr3Level)level is Tr3Level.Reunion,
+            _ => throw new ArgumentOutOfRangeException(nameof(CurrentActiveGame), CurrentActiveGame, "Unknown Game"),
```

Comment:
`IsFirstLevel` is indented but this isn't. You should probably remove this method anyway.

Reply:
That's odd, I see them both as indented in Visual Studio. But yes, I'll remove it.